x-pack/plugins/security_solution/server/lib/telemetry/endpoint_metadata_telemetry.ts

```diff
@@ -0,0 +1,124 @@
+import { AGENT_SAVED_OBJECT_TYPE } from './../../../../ingest_manager/common/constants/agent';
+import { AgentSOAttributes } from '../../../../ingest_manager/common';
+
+export interface EndpointMetadataOSTelemtry {
+  name: string;
+  version: string;
+  count: number;
+}
+
+export interface EndpointPoliciesTelemetry {
+  malware: {
+    success: number;
+    warning: number;
+    failure: number;
+  };
+}
+
+export interface IngestManagerLocalMetadata extends AgentMetadata {
+  host: {
+    id: string;
+  };
+  os: {
+    name: string;
+    version: string;
+    full: string;
+  };
+}
+
+export interface EndpointMetadataTelemetry {
+  endpoints: {
+    installed: number;
+    last_24_hours: number;
+  };
+  os: Record<string, EndpointMetadataOSTelemtry>;
+  policies: EndpointPoliciesTelemetry;
+}
+
+export const getDefaultEndpointMetadataTelemetry = (): EndpointMetadataTelemetry => ({
+  endpoints: { installed: 0, last_24_hours: 0 },
+  os: {} as Record<string, EndpointMetadataOSTelemtry>,
+  policies: {
+    malware: {
+      success: 0,
+      warning: 0,
+      failure: 0,
+    },
+  },
+});
+
+/**
+ * ---Functionality Tracking---
+ * Number of active endpoints (in progress)
+ *  - Successfully installed (done)
+ *  - Has sent response within last 24 hours (best way to go about this? May not have info?)
+ * See the list by customer (provided by telemetry under license.issued_to) Can most likely pull those details in the dashboards
+ * OS (done)
+ *  - Type (done)
+ *  - Version (done)
+ * Geolocation of endpoints (slated for 7.10): Query the agent based on the agent info in the metadata to retrieve this
+ * ---------------------------------
+ * RE Performance / usage collection timeout from Tina Helligers
+ * We have a timeout on waiting for collectors to report they are ready before issuing a bulkFetch method that essentially calls the fetch method for all the collectors registered.
+ * This timeout defaults to 60s. If you suspect that your fetch method is going to take too long (meaning data for that collector won't be reported),
+ * you could follow a similar approach as what we do for oss_telemetry where we use the task manager to run on a daily basis.
+ */
+
+const ENDPOINT_PACKAGE_CONSTANT = 'system'; // TODO: replace with 'endpoint' when I have endpoint package working remotely
+
+export const getEndpointMetadataTelemetryFromFleet = async (
+  savedObjectsClient: ISavedObjectsRepository
+) => {
+  const { saved_objects: endpointAgents } = await savedObjectsClient.find<AgentSOAttributes>({
+    type: AGENT_SAVED_OBJECT_TYPE,
+    fields: ['packages', 'last_checkin', 'local_metadata'],
+    filter: `${AGENT_SAVED_OBJECT_TYPE}.attributes.packages: ${ENDPOINT_PACKAGE_CONSTANT}`,
+  });
+  const uniqueHostIds: Set<string> = new Set();
+  const endpointTelemetry = getDefaultEndpointMetadataTelemetry();
+  endpointTelemetry.endpoints.installed = endpointAgents.length; // All agents who have the endpoint registered
```

Comment:
The agent references installed packages in the `packages` field here. So when we filter for it, this should give us all of the agents that have installed the endpoint. Right now I'm using `system` since I haven't been able to get the endpoint successfully installed via the agent 😞. @nchaulet , can you confirm my previous statement is accurate?

Reply:
Actually just saw a unique case where I enrolled a second agent on an endpoint and got 2 installed instead of one. Using the unique hosts id's now to track the count to avoid this issue